clang/runtime/dpct-rt/include/dpct/group_utils.hpp

```diff
@@ -842,16 +842,119 @@ class workgroup_load {
+  // storage
+  size_t linear_tid = item.get_local_linear_id();
+  OutputIteratorT workitem_itr = block_itr + linear_tid;
+  size_t group_work_items = item.get_local_range().size();
+#pragma unroll
+  for (uint32_t idx = 0; idx < ITEMS_PER_WORK_ITEM; idx++) {
+    workitem_itr[(idx * group_work_items)] = items[idx];
+  }
+}
+
+/// Stores a subgroup-striped arrangement of work items linear segment of items.
+// Created as free function until exchange mechanism is
+// implemented.
+// To-do: inline this function with BLOCK_STORE_WARP_TRANSPOSE mechanism
+template <size_t ITEMS_PER_WORK_ITEM, typename InputT, typename OutputIteratorT,
+          typename Item>
+__dpct_inline__ void
+store_subgroup_striped(const Item &item, OutputIteratorT block_itr,
+                       InputT (&items)[ITEMS_PER_WORK_ITEM]) {
+
+  // This implementation does not take in account range storing across
+  // workgroup items To-do: Decide whether range storing is required for group
+  // loading
+  // This implementation loads linear segments into warp striped arrangement.
+  auto sub_group = item.get_subgroup();
+  uint32_t subgroup_offset = sub_group.get_local_linear_id();
+  uint32_t subgroup_size = sub_group.get_local_linear_range();
+  uint32_t subgroup_idx = sub_group.get_group_linear_id();
+  uint32_t initial_offset =
+      (subgroup_idx * ITEMS_PER_WORK_ITEM * subgroup_size) + subgroup_offset;
+  OutputIteratorT workitem_itr = block_itr + initial_offset;
+#pragma unroll
+  for (uint32_t idx = 0; idx < ITEMS_PER_WORK_ITEM; idx++) {
+    workitem_itr[(idx * subgroup_size)] = items[idx];
+  }
+}
+
+// template parameters :
+// ITEMS_PER_WORK_ITEM: size_t variable controlling the number of items per
+// thread/work_item
+// ALGORITHM: store_algorithm variable controlling the type of store operation.
+// InputT: type for input sequence.
+// OutputIteratorT:  output iterator type
+// Item : typename parameter resembling sycl::nd_item<3> .
+template <size_t ITEMS_PER_WORK_ITEM, store_algorithm ALGORITHM,
+          typename InputT, typename OutputIteratorT, typename Item>
+class workgroup_store {
+public:
+  static size_t get_local_memory_size(size_t group_work_items) { return 0; }
+  workgroup_store(uint8_t *local_memory) : _local_memory(local_memory) {}
+
+  __dpct_inline__ void store(const Item &item, OutputIteratorT block_itr,
+                             InputT (&items)[ITEMS_PER_WORK_ITEM]) {
+
+    if constexpr (ALGORITHM == store_algorithm::BLOCK_STORE_DIRECT) {
+      store_blocked<ITEMS_PER_WORK_ITEM>(item, block_itr, items);
+    } else if constexpr (ALGORITHM == store_algorithm::BLOCK_STORE_STRIPED) {
+      store_striped<ITEMS_PER_WORK_ITEM>(item, block_itr, items);
+    }
+  }
+
+private:
+  // local_memory is a placeholder ,currently unused, as no operations use
+  // extra memory but placed here to make migrations easier. 
+  // For future exchange operations might be necessary
```

Comment:
```suggestion
  // _local_memory is a placeholder, currently unused, as no operations use
  // extra memory but placed here in case future strategies or 
  // implementations require temporary memory.
```